### DISCUSSION TITLE
BIS schedual query

### DISCUSSION THREAD

**Discussion by salah-walid** [iTwin/itwinjs-core#3876] [Q&A] [answered]
Hi,

In order to get the full data of schedule from the BIM, what would you say is the correct SQL query to get it ?

**Comment by pmconne** [comment, marked as answer]
Are you asking about [schedule scripts](https://www.itwinjs.org/reference/core-common/displaystyles/renderschedule/)?
Can you explain what you mean by "full data"? The iModel contains only the information required to render the schedule.

An iModel can contain any number of schedule scripts. These can be stored either in the JSON properties of a [DisplayStyle](https://www.itwinjs.org/reference/core-backend/viewdefinitions/displaystyle/) or on a [RenderTimeline](https://www.itwinjs.org/reference/core-backend/elements/rendertimeline/) element.

In what context are you trying to obtain the schedule script(s)? Do you have a viewport and want to get the script specifically associated with that viewport? Or...?

**Reply by salah-walid** [reply]
What i am looking for is a detailed schedual, with all the task name, duration...
With schedule script you don't get the task names for example, what i want to get is similar to when you make a request to `https://synchro-imodelsync-san.bentley.com/sv1.0/Repositories/Project--${projectId}/Schedule/Task?$select=*,TaskSuccessor-forward-Task.$id,TaskSuccessor.*`

**Reply by pmconne** [reply]
That information is not stored in the iModel.

**Reply by salah-walid** [reply]
I suppose there is another database storing that data, if so is there a way to access it ?

**Reply by pmconne** [reply]
I checked with the SYNCHRO team. Here is their response:

> Tthere is not such API available yet. I don’t know when it would be available.
> synchro-imodelsync-XXX.bentley.com is internal API, and not supported for external users.

**Reply by ayushibasaiwal** [reply]
@salah-walid  If your query is resolved, please mark this reply as an answer, or in case you need any help, please ask.
